File: baba-chatbot-python/src/baba_chatbot/domain/value_objects/vehicle_query.py
"""
Value Object que representa una consulta de búsqueda de vehículos.
"""
from dataclasses import dataclass
from typing import Optional


@dataclass
class VehicleQuery:
    """
    Value Object que encapsula los criterios de búsqueda de vehículos.
    """
    brand: Optional[str] = None
    model: Optional[str] = None
    min_price: Optional[float] = None
    max_price: Optional[float] = None
    min_year: Optional[int] = None
    max_year: Optional[int] = None
    max_mileage: Optional[int] = None

    def has_any_filter(self) -> bool:
        """
        Verifica si la consulta tiene algún filtro aplicado.
        
        Returns:
            True si al menos un filtro está definido, False en caso contrario
        """
        return any([
            self.brand is not None,
            self.model is not None,
            self.min_price is not None,
            self.max_price is not None,
            self.min_year is not None,
            self.max_year is not None,
            self.max_mileage is not None
        ])

File: baba-chatbot-net/src/Baba.Chatbot.Domain/ValueObjects/VehicleQuery.cs
namespace Baba.Chatbot.Domain.ValueObjects;

/// <summary>
/// Objeto de valor que representa los criterios de búsqueda para filtrar vehículos
/// Inmutable por ser un record
/// </summary>
public record VehicleQuery
{
    public string? Brand { get; init; }
    public string? Model { get; init; }
    public int? MinYear { get; init; }
    public int? MaxYear { get; init; }
    public decimal? MinPrice { get; init; }
    public decimal? MaxPrice { get; init; }
    public int? MaxMileage { get; init; }
    public string? Transmission { get; init; }
    public string? FuelType { get; init; }
    public string? BodyType { get; init; }
    public List<string> RequiredFeatures { get; init; } = new();

    /// <summary>
    /// Verifica si se especificó al menos un filtro de búsqueda
    /// </summary>
    public bool HasAnyFilter() =>
        Brand != null ||
        Model != null ||
        MinYear.HasValue ||
        MaxYear.HasValue ||
        MinPrice.HasValue ||
        MaxPrice.HasValue ||
        MaxMileage.HasValue ||
        Transmission != null ||
        FuelType != null ||
        BodyType != null ||
        RequiredFeatures.Count > 0;
}

